File: src/Vehicle/src/CarList/CarList.java
package CarList;

import Car.Car;

public class CarList {
  public void inputCar(Car list[], int n) {
    for (int i = 0; i < n; i++) {
      list[i] = new Car();
      System.out.println("Car " + "[" + (i + 1) + "]" + ": ");
      list[i].input();
    }
  }

  public void outputCar(Car list[], int n) {
    for (int i = 0; i < n; i++) {
      System.out.print("Car " + "[" + (i + 1) + "]" + ": ");
      list[i].output();
    }
  }

  public void sort(Car list[], int n) {
    for (int i = 0; i < n; i++) {
      for (int j = i + 1; j < n; j++) {
        if (list[i].getPriceWithVAT() > list[j].getPriceWithVAT()) {
          Car temp = new Car();
          temp = list[i];
          list[i] = list[j];
          list[j] = temp;
        }
      }
    }
    System.out.println("Sorting list of cars increases by the price with VAT: ");
    for (int i = 0; i < n; i++) {
      list[i].output();
    }
  }

  public float sumOfCash(Car list[], int n) {
    float sum = 0;
    for (int i = 0; i < n; i++) {
      sum += list[i].getPriceWithVAT();
    }
    return sum;
  }

  public void maxCash(Car list[], int n) {
    float max = list[0].getPriceWithoutVAT();
    for (int i = 0; i < n; i++) {
      if (list[i].getPriceWithoutVAT() > max) {
        max = list[i].getPriceWithoutVAT();
      }
    }
  }

  public void searchByName(String x, Car list[], int n) {
    int check = 0;
    for (int i = 0; i < n; i++) {
      if (x.equals(list[i].getSource())) {
        check = 1;
        list[i].output();
      }
    }
    if (check == 0) {
      System.out.print("Dont have any car");
    }
  }
}

File: 0610/CarList.cpp
#include <iostream>
#include "CarList.h"
using namespace std;

void CarList::inputCar(int n) {
  for(int i = 0; i < n; i++) {
    cout << "Car " << (i + 1) << ": \n";
    car[i].input();
  }
}
void CarList::outputCar(int n) {
  for(int i = 0; i < n; i++) {
    cout << "\nCar " << (i + 1) << ": ";
    car[i].output();
  }
}

void CarList::sort(int n) {
  for(int i = 0; i < n; i++) {
    for(int j = i + 1; j < n; j++) {
      if(car[i].priceWithVAT > car[j].priceWithVAT) {
        Car temp = car[i];
        car[i] = car[j];
        car[j] = temp;
      }
    }
  }

  cout << "\nSorting list of cars increases by the price with VAT (priceWithVAT).\n";
  outputCar(2);
}

void CarList::sumOfCash(int n) {
  float sum = 0;
  for(int j = 0; j < n; j++) {
      sum += car[j].priceWithVAT;
  }
  cout << "\nCalculate the total of the price with VAT of the list of cars = " << sum;
}

void CarList::maxCash(int n) {
  float max = 0;
  for(int j = 0; j < n; j++) {
      if(car[j].priceWithVAT > max) max = car[j].priceWithVAT;
  }

  cout << "\nDisplay information of the cars that have the highest price without VAT: \n";
  for(int j = 0; j < n; j++) {
      if(car[j].priceWithVAT == max) car[j].output();
  }
}

void CarList::searchByName(int n) {
  string carName;
  cout << "\nSearch the car's information by any manufacturer name entered from the keyboard: ";
  cin.ignore();
  getline(cin, carName);
  for(int j = 0; j < n; j++) {
    if(car[j].manufacturer == carName) {
      car[j].output();
      return;
    }
    else {
      cout << "No car with name";
      return;
    }
  }
}
